src/firefly/plans/xafs_scan.py

```diff
@@ -151,6 +177,43 @@ def update_wavenumber_energy(self):
             else:
                 self.update_line_edit_value(line_edit, func)
 
+    def update_total_time(self):
+        weight = self.weight_spinbox.value()
+        exposure_time = self.exposure_time_spinbox.value()
+
+        # prevent invalid inputs such as nan
+        try:
+            start = round(float(self.start_line_edit.text()), float_accuracy)
+            stop = round(float(self.stop_line_edit.text()), float_accuracy)
+            step = round(float(self.step_line_edit.text()), float_accuracy)
+
+        # when the round doesn't work for nan values
+        except ValueError:
+            self.kErange = []
+            start, stop, step = float("nan"), float("nan"), float("nan")
+
+        finally:
+            if self.k_space_checkbox.isChecked():
```

Comment:
You don't need the ``finally`` here. These can just run outside of the ``try…except`` block.

Reply:
resolved